### PR TITLE
HOSTEDCP-1788: Create seperate RGs for NSG and VNet

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:
Creates seperate RGs for NSG and VNet

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
[HOSTEDCP-1788](https://issues.redhat.com/browse/HOSTEDCP-1788)

**Checklist**
- [X] Subject and description added to both, commit and PR.
- [X] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by stevekuznetsov — COMMENTED** [review]
Are you able to set up a unit test similar to the `TestCreateCluster` for Azure that I did, so that changes to this part of the codebase are tested?

**Comment by enxebre** [comment]
/hold
until there's managed=aro e2e in place

**Comment by bryan-cox** [comment]
/unhold
We have e2e-aks now

File: cmd/infra/azure/create.go [hunk 1]
```diff
@@ -244,7 +261,7 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 }
 
 // createResourceGroup creates the Azure resource group used to group all Azure infrastructure resources
-func createResourceGroup(ctx context.Context, o *CreateInfraOptions, azureCreds azcore.TokenCredential, subscriptionID string) (string, string, string, error) {
+func createResourceGroup(ctx context.Context, o *CreateInfraOptions, azureCreds azcore.TokenCredential, rgName, subscriptionID string) (string, string, string, error) {
```
Comment (enxebre) [inline cmd/infra/azure/create.go:264]: Is the comment of this signature true any more? can we at minimum update that to reflect reality and how this parameter impacts the output?

Comment (Patryk-Stefanski) [reply cmd/infra/azure/create.go:267]: Done

File: cmd/infra/azure/destroy.go [hunk 2]
```diff
@@ -73,13 +77,29 @@ func (o *DestroyInfraOptions) Run(ctx context.Context, logger logr.Logger) error
 		return fmt.Errorf("failed to create new resource groups client: %w", err)
 	}
 
-	destroyFuture, err = resourceGroupClient.BeginDelete(ctx, o.GetResourceGroupName(), nil)
-	if err != nil {
-		return fmt.Errorf("failed to start deletion: %w", err)
+	var resourceGroups []string
+	resourceGroups = append(resourceGroups, o.ResourceGroupName)
```
Comment (fxierh) [inline cmd/infra/azure/destroy.go:81]: Could we leverage o.GetResourceGroupName() so hypershift destroy infra can be run without needing to specify --resource-group-name

File: cmd/infra/azure/create.go [hunk 3]
```diff
@@ -159,8 +159,16 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 		result.SecurityGroupID = o.NetworkSecurityGroupID
 		l.Info("Using existing network security group", "ID", result.SecurityGroupID)
 	} else {
+		// Create a resource group for network security group
+		nsgResourceGroupName := o.Name + "-nsg"
+		_, nsgRG, msg, err := createResourceGroup(ctx, o, azureCreds, nsgResourceGroupName, subscriptionID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create resource group for network security group: %w", err)
+		}
+		l.Info(msg, "name", nsgResourceGroupName)
```
[the remark this answers is not in the document]
Comment (Patryk-Stefanski) [reply cmd/infra/azure/create.go:168]: This only logs that a resource group has been successfully created do we need to log the whole ResourceGroup object ?

[the remark this answers is not in the document]
Comment (fxierh) [reply cmd/infra/azure/create.go:168]: Sorry for the confusion. I meant nsgRG != nsgResourceGroupName and nsgRG is the actual name of the RG created.

**Pull request merged**